sqe-database-access/Models/InterpretationAttribute.cs

```diff
@@ -8,6 +8,20 @@ public class CharAttribute
         public string attributeString { get; set; }
         public uint signInterpretationAttributeAuthor { get; set; }
         public float value { get; set; }
+        
+        // The override for Equals and GetHashCode methods here enable the
+        // HashSet nextSignInterpretations of the SignInterpretation object
+        // to ensure that no duplicate values will be inserted into the set.
+        public override bool Equals(object obj)
+        {
+            return obj is CharAttribute q 
+                   && q.interpretationAttributeId == this.interpretationAttributeId;
+        }
+
+        public override int GetHashCode()
+        {
+            return unchecked((int) this.interpretationAttributeId);
```

Comment:
This had me going to read about `unchecked`, since I've never used it before. Turns out `(int)u` is enough, no need for unchecked. No need to change the code, just something that's useful to know:
https://stackoverflow.com/a/1131851/871910